server/src/main/java/hu/flowacademy/epsilon/myfavoriteexpert/service/UserService.java

```diff
@@ -79,4 +113,71 @@ public User saveLanguage(String language) {
         }
         return userRepository.save(user);
     }
+    public List<User> findBestMatchedUserByName(String searchParams) {
+        Pageable pageable = PageRequest.of(0,1);
+        searchParams = searchParams.replaceAll("_"," ");
+        return userRepository.findBestMatchesUser(searchParams,pageable).getContent();
+    }
+    public List<Expert> findExpertsByUser(String searchParams) {
+        Pageable pageable = PageRequest.of(0,1);
+        searchParams = searchParams.replaceAll("_"," ");
+        List<User> users = userRepository.findExpertsByUser(searchParams,pageable)
+                .getContent();
+        List<Expert> experts = new ArrayList<>();
+        if (users.size() > 0) {
+
+                 experts = users.get(0)
+                    .getExperts()
+                    .stream()
+                    .map(expertId -> expertRepository.findById(expertId).orElse(null))
+                    .collect(Collectors.toList());
+        }
+        return experts;
+    }
+    public List<Expert> findAllExperts(UUID userId) {
+        User user = userRepository.findById(userId).orElse(null);
+        List<Expert> experts;
+        if (user != null && user.getExperts() != null) {
+            experts = user.getExperts().stream()
+                    .map(expertid -> expertRepository
+                            .findById(expertid).orElse(null))
+                    .collect(Collectors.toList());
+            experts = expertService.setExpertDistanceAndLikes(experts);
+            return experts;
+        }
+        throw new RuntimeException("ID INVALID");
+    }
+    public List<Expert> findUsersExpertsIntersection(UUID id1) {
+        User user1 = userRepository.findById(id1).orElse(null);
+        User user2 = userRepository.findById(getCurrentUserId()).orElseThrow(RuntimeException::new);
+        Set<Expert> experts1 = new HashSet<>();
+        Set<Expert> experts2;
+        List<Expert> expertsIntersection = new ArrayList<>();
+        if (user1 != null && user1.getExperts() != null && user2 != null && user2.getExperts() != null) {
+            experts1 = user1.getExperts().stream()
+                    .map(expertid -> expertRepository
+                            .findById(expertid).orElse(null))
+                    .collect(Collectors.toSet());
+            experts2 = user2.getExperts().stream()
+                    .map(expertid -> expertRepository
+                            .findById(expertid).orElse(null))
+                    .collect(Collectors.toSet());
+            experts1.retainAll(experts2);
+            experts1 = expertService.setExpertDistanceAndLikes(experts1);
+            expertsIntersection = Lists.newArrayList(experts1);
+        }
+
+        return expertsIntersection;
```

Comment:
Itt meg egy List.of()-ot. Ebben az esetben nem lesz érték

Reply:
fixed